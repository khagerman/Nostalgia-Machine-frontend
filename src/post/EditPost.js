import React, { useState, useContext } from "react";
import Alert from "../common/Alert";
import NostalgiaApi from "../api";
import { useHistory } from "react-router-dom";
import UserContext from "../auth/UserContext";
import { ToastContainer, toast } from "react-toastify";
import { Container, Card, FormGroup, TextField, Button } from "@mui/material";
import { Formik, Form, Field, ErrorMessage } from "formik";
import { object, string } from "yup";
export default function PopUpEditPost({
  id,
  toggle,
  title,
  url,
  updatedPost,
  updatePost,
}) {
  const { currentUser, setCurrentUser } = useContext(UserContext);

  let handleClick = () => {
    toggle();
  };

  let username = currentUser?.username;

  async function handleSubmit(values) {
    try {
      let post = await NostalgiaApi.patchPost(id, {
        ...values,
      });
      toggle();
      let currentUser = await NostalgiaApi.getUser(username);
      setCurrentUser(currentUser);
      updatePost(post);
    } catch (errors) {
      console.log(errors);
    }
  }

  const PostSchema = object({
    url: string().required().url(),
    title: string().required().min(1).max(100),
  });
  return (
    <Card sx={{ maxWidth: 345 }}>
      <span className="close" onClick={handleClick}>
        &times;
      </span>
      <Formik
        initialValues={{
          url: url,
          title: title,
        }}
        validationSchema={PostSchema}
        onSubmit={(values) => {
          handleSubmit(values);
        }}
      >
        {({ errors, touched }) => (
          <Form>
            <FormGroup>
              <label>Image URL</label>
              <Field
                name="url"
                className="form-control"
                label="Image URL"
                as={TextField}
              />
              {errors.url && touched.url ? (
                <div style={{ color: "red" }}>{errors.url}</div>
              ) : null}
            </FormGroup>
            <FormGroup>
              <label>Post Text</label>
              <Field
                name="title"
                id="title"
                className="form-control"
                as={TextField}
              />
              {errors.title && touched.title ? (
                <div style={{ color: "red" }}>{errors.title}</div>
              ) : null}
            </FormGroup>
            <Button
              variant="contained"
              type="submit"
              className="btn btn-primary float-right"
              onSubmit={handleSubmit}
            >
              Submit
            </Button>
          </Form>
        )}
      </Formik>
    </Card>
  );
}
