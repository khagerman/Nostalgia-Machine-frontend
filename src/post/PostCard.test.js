import React from "react";
import { render } from "@testing-library/react";
import EditPost from "./EditPost";

it("matches snapshot", function () {
  const { asFragment } = render(<EditPost />);
  expect(asFragment()).toMatchSnapshot();
});
