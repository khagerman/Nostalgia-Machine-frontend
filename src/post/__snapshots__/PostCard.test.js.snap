// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root Card css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <a
        class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
        href="/post/undefined"
        tabindex="0"
      >
        <img
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img img-fluid css-o69gx8-MuiCardMedia-root"
        />
        <h3
          class="title m-2"
        />
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <i
          class="far fa-heart"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
